src/features/smSystem/suppliersOrders/components/AddOrderModal/Modal.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { GridRowParams } from '@mui/x-data-grid';
+import { SetStateAction, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Pages } from '../../../../../utils';
+import {BasicModalProps} from '../../../app/types/index';
+import { useCreateOrder } from '../../api/useCreateOrder';
+import { useGetSuppliers } from '../../api/useGetSuppliers';
+
+import BranchesTable from './BranchesTable';
+import SuppliersTable from './SuppliersTable';
+
+
+const AddOrderModal = ({ open, handleClose }: BasicModalProps) => {
+  const [page, setPage] = useState(0);
+  const [name, setName] = useState<string>('');
+  const [nameToSearch, setNameToSearch] = useState<string>('');
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+
+  const {
+    data,
+    isLoading,
+    isFetchingNextPage,
+    fetchNextPage,
+    isError,
+    name: searchedName,
+  } = useGetSuppliers(nameToSearch);
+  const [selectedSupplier, setSelectedSupplier] = useState<number | null>(null);
+  const [selectedBranches, setSelectedBranches] = useState<number[]>([]);
```

Comment:
Keep state together, in components, keep similar things in groups, like: general hooks, states, local hooks, etc, sometimes the order is important and you can deviate from the rule